edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/deployment/KubernetesDeploymentMapper.cs

```diff
@@ -132,9 +132,10 @@ V1PodTemplateSpec GetPod(string name, IModuleIdentity identity, KubernetesModule
                 .Select(pullSecretName => new V1LocalObjectReference(pullSecretName))
                 .ToList();
 
-            V1PodSecurityContext securityContext = this.runAsNonRoot
-                ? new V1PodSecurityContext { RunAsNonRoot = true, RunAsUser = 1000 }
-                : null;
+            V1PodSecurityContext securityContext = module.Config.CreateOptions.SecurityContext.GetOrElse(
+                () => this.runAsNonRoot
```

Comment:
I am struggling to see the value of having us store a default for runAsNonRoot in every V1PodSecurityContext and ignore it when module CreateOptions don't set any default runAsNonRoot. Our agent will not error out or warn anybody about this happening, making troubleshooting a pain for operators / module developers. Can we stay consistent with policy always and override the module CreateOption runAsNonRoot setting, when the helm chart runAsNonRoot policy is true?

Reply:
The main idea we discussed: we don't want to dictate the user how to create their modules. We should allow a user to run privileged containers if it what he wants. If he opt-in `runAsNonRoot: true` in helm chart we should put this policy. If somehting is specified in CreateOptions section it will override whatever defaults we have.